conformance_tests/tools/debug/src/test_debug.cpp

```diff
@@ -1376,6 +1377,143 @@ void zetDebugReadWriteRegistersTest::run_read_write_registers_test(
   }
 }
 
+void zetDebugReadWriteRegistersTest::run_read_registers_test(
+    std::vector<ze_device_handle_t> &devices, bool use_sub_devices) {
+  for (auto &device : devices) {
+    print_device(device);
+    if (!is_debug_supported(device))
+      continue;
+
+    synchro->clear_debugger_signal();
+    debugHelper = launch_process(LONG_RUNNING_KERNEL_INTERRUPTED_SCRATCH,
+                                 device, use_sub_devices);
+
+    zet_debug_event_t module_event;
+    attach_and_get_module_event(debugHelper.id(), synchro, device, debugSession,
+                                module_event);
+
+    if (module_event.flags & ZET_DEBUG_EVENT_FLAG_NEED_ACK) {
+      LOG_DEBUG << "[Debugger] Acking event: "
+                << lzt::debuggerEventTypeString[module_event.type];
+      lzt::debug_ack_event(debugSession, &module_event);
+    }
+
+    uint64_t gpu_buffer_va = 0;
+    synchro->wait_for_application_signal();
+    if (!synchro->get_app_gpu_buffer_address(gpu_buffer_va)) {
+      FAIL() << "[Debugger] Could not get a valid GPU buffer VA";
+    }
+    synchro->clear_application_signal();
+
+    zet_debug_memory_space_desc_t memorySpaceDesc;
+    memorySpaceDesc.type = ZET_DEBUG_MEMORY_SPACE_TYPE_DEFAULT;
+    int sizeToRead = 512;
+    uint8_t *kernel_buffer = new uint8_t[sizeToRead];
+    // set buffer[0] to 0 to break the loop. See debug_loop_slm.cl
+    kernel_buffer[0] = 0;
+    memorySpaceDesc.address = gpu_buffer_va;
+
+    ze_device_thread_t device_threads = {};
+    device_threads.slice = UINT32_MAX;
+    device_threads.subslice = UINT32_MAX;
+    device_threads.eu = UINT32_MAX;
+    device_threads.thread = UINT32_MAX;
+
+    LOG_INFO << "[Debugger] Stopping all device threads";
+    // give time to app to launch the kernel
+    std::this_thread::sleep_for(std::chrono::seconds(6));
+    lzt::debug_interrupt(debugSession, device_threads);
+
+    std::vector<ze_device_thread_t> stopped_threads;
+    if (!find_stopped_threads(debugSession, device, device_threads, true,
+                              stopped_threads)) {
+      delete[] kernel_buffer;
+      FAIL() << "[Debugger] Did not find stopped threads";
+    }
+
+    bool is_heapless_mode = false;
```

Comment:
Its a code redundancy in our CTS. could you please add a method to determine is_heapless_mode. And at all the places in other CTS we could use that

Reply:
Done